docs/data/material/getting-started/templates/checkout/PaymentForm.js

```diff
@@ -1,65 +1,124 @@
 
+import CreditCardRoundedIcon from '@mui/icons-material/CreditCardRounded';
+import SimCardRoundedIcon from '@mui/icons-material/SimCardRounded';
+
 export default function PaymentForm() {
+  const [cardNumber, setCardNumber] = React.useState('');
+  const [cvv, setCvv] = React.useState('');
+  const [expirationDate, setExpirationDate] = React.useState('');
+
+  const handleCardNumberChange = (event) => {
+    const value = event.target.value.replace(/\D/g, '');
+    const formattedValue = value.replace(/(\d{4})(?=\d)/g, '$1 ');
+    if (value.length <= 16) {
+      setCardNumber(formattedValue);
+    }
+  };
+
+  const handleCvvChange = (event) => {
+    const value = event.target.value.replace(/\D/g, '');
+    if (value.length <= 3) {
+      setCvv(value);
+    }
+  };
+
+  const handleExpirationDateChange = (event) => {
+    const value = event.target.value.replace(/\D/g, '');
+    const formattedValue = value.replace(/(\d{2})(?=\d{2})/, '$1/');
+    if (value.length <= 4) {
+      setExpirationDate(formattedValue);
+    }
+  };
+
   return (
     <React.Fragment>
-      <Typography variant="h6" gutterBottom>
-        Payment method
-      </Typography>
-      <Grid container spacing={3}>
-        <Grid item xs={12} md={6}>
-          <TextField
-            required
-            id="cardName"
-            label="Name on card"
-            fullWidth
-            autoComplete="cc-name"
-            variant="standard"
-          />
-        </Grid>
-        <Grid item xs={12} md={6}>
-          <TextField
-            required
-            id="cardNumber"
-            label="Card number"
-            fullWidth
-            autoComplete="cc-number"
-            variant="standard"
-          />
-        </Grid>
-        <Grid item xs={12} md={6}>
-          <TextField
-            required
-            id="expDate"
-            label="Expiry date"
-            fullWidth
-            autoComplete="cc-exp"
-            variant="standard"
-          />
-        </Grid>
-        <Grid item xs={12} md={6}>
-          <TextField
-            required
-            id="cvv"
-            label="CVV"
-            helperText="Last three digits on signature strip"
-            fullWidth
-            autoComplete="cc-csc"
-            variant="standard"
-          />
-        </Grid>
-        <Grid item xs={12}>
-          <FormControlLabel
-            control={<Checkbox color="secondary" name="saveCard" value="yes" />}
-            label="Remember credit card details for next time"
+      <Box
+        sx={{
+          display: 'flex',
+          justifyContent: 'center',
+          mb: 4,
+        }}
+      >
+        <Box
+          sx={{
+            display: 'flex',
+            flexDirection: 'column',
+            justifyContent: 'space-between',
+            p: 3,
+            height: { xs: '250px', sm: '350px' },
+            width: { xs: '375px', sm: '525px' },
+            borderRadius: '20px',
+            backdropFilter: 'blur(35px)',
```

Comment:
For the background filter to work well, there should be some sort of background image in the container so that it blurs that; otherwise, there's not much visual impact. Something like:

<img width="350" alt="Screenshot 2024-02-27 at 17 11 28" src="https://github.com/mui/material-ui/assets/67129314/d3f2b483-4d0e-4abe-86cd-029bb971927a">

In this quick example, I added a random background image (`background-image: url(https://source.unsplash.com/Q7PclNhVRI0)`) and then used `outline` for the border instead of `border` so it outlines the outside of the container, not inside.